tests/features/schema-generator/type-infering-from-default.test.ts

```diff
@@ -0,0 +1,36 @@
+import { Entity, ManyToOne, PrimaryKey, Property } from '@mikro-orm/core';
+import { MikroORM } from '@mikro-orm/better-sqlite';
+
+@Entity()
+export class Author {
+
+  @PrimaryKey()
+  id!: number;
+
+  @Property({ default: 0 })
+  age=0;
```

Comment:
```suggestion
  age = 0;
```

please follow this convention, there is more places with this 

and please dont rebase once you get a review, i cant see what changes you make

Reply:
Current lint didn't work in my environment, and it looks like adding this rule works.
Or is there other rule that should have worked?

```
"space-infix-ops": ["error", {"int32Hint": true}]
```

Reply:
yes, but without the `int32Hint` option i guess

Reply:
https://github.com/mikro-orm/mikro-orm/commit/428e41f8b69155cb41c1ba6ebd93358653d78ed9

Reply:
dont forget to test this works with the inferred defaults:

```ts
  @Property()
  age = 0;
```

this needs to work too. i'd test numbers, strings, dates, buffers, booleans, and also some objects to test the fallback, as i am still not sure about that, if the default is object type, we might want to respect that and infer JSON property type

Reply:
Sure, I will include types you mentioned. Thanks :)

Reply:
The reflect-metadata library is not inferring the default value from the line below like ts-morph does.
```let type = Reflect.getMetadata('design:type', meta.prototype, prop.name);```
Is there a way to get default value from reflect-metadata, or do you have any idea on this?

Reply:
The whole point of this PR is to support that, because `reflect-metadata` is not capable of that.

Reply:
Okay. So we have two problems. One is to infer type from `PropertyOptions`, which we can do it in `ReflectMetadataProvider`. The Other one is to infer type when assign default value with `=`. It seem a little tricky. Will it be appropriate to infer when creating `MetadataStorage.metadata`?

Reply:
Not really, I'd say there is just one problem. I guess I will just take over if you don't mind, maybe the problem is harder than I envisioned.

Reply:
Okay. Thank you.

Reply:
It's indeed more complicated than I thought. The problem itself was rather easy (the trick is to move the `initDefaultValue` call before the metadata provider so the inference happens early, before the default string type is added), but when I started adjusting existing entities to test this there too, many tests were failing because of connected issues, like the `ref` helper not working with null values, causing constructors to fail.